{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9fde5b6b-5ce9-4f5e-a27e-65f28755b505",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-08-26 16:23:31.656158: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory\n",
      "2021-08-26 16:23:31.656212: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.\n"
     ]
    }
   ],
   "source": [
    "from monet_boys.data import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "6c756b6f-9a4a-44cf-838d-6e6758d6cdd9",
   "metadata": {},
   "outputs": [],
   "source": [
    "example_cezanne = next(iter(cezanne_ds)).numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "5f075545-71f2-4a37-ac0b-0e093a46d9f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "example_cezanne = (example_cezanne * 127.5 + 127.5).astype(np.uint8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "dff3823d-c2f1-449f-aba9-b3f2c57ca1fc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "decode_image\n",
      "read_image\n",
      "load_jpg_dataset\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "Cannot handle this data type: (1, 1, 3), <f4",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/.pyenv/versions/3.8.6/envs/lewagon/lib/python3.8/site-packages/PIL/Image.py\u001b[0m in \u001b[0;36mfromarray\u001b[0;34m(obj, mode)\u001b[0m\n\u001b[1;32m   2771\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2772\u001b[0;31m             \u001b[0mmode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrawmode\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_fromarray_typemap\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mtypekey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2773\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: ((1, 1, 3), '<f4')",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_22352/1216274019.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mimg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mexample_cezanne\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mexample_cezanne\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mexample_cezanne\u001b[0m \u001b[0;34m*\u001b[0m \u001b[0;36m127.5\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m127.5\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0muint8\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0mimg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mPIL\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mImage\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfromarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mimg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.pyenv/versions/3.8.6/envs/lewagon/lib/python3.8/site-packages/PIL/Image.py\u001b[0m in \u001b[0;36mfromarray\u001b[0;34m(obj, mode)\u001b[0m\n\u001b[1;32m   2772\u001b[0m             \u001b[0mmode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrawmode\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_fromarray_typemap\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mtypekey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2773\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2774\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Cannot handle this data type: %s, %s\"\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mtypekey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2775\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2776\u001b[0m         \u001b[0mrawmode\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmode\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: Cannot handle this data type: (1, 1, 3), <f4"
     ]
    }
   ],
   "source": [
    "cezanne_ds = load_jpg_dataset(CEZANNE_FILENAMES)\n",
    "example_cezanne = next(iter(cezanne_ds)).numpy()\n",
    "img = example_cezanne\n",
    "example_cezanne = (example_cezanne * 127.5 + 127.5).astype(np.uint8)\n",
    "img=img[0]\n",
    "img = PIL.Image.fromarray(img)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "19ee7553-b989-48d0-88b2-261a66627db0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.image.AxesImage at 0x7f17f0325e50>"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAADIAAAD8CAYAAAA13fd1AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8rg+JYAAAACXBIWXMAAAsTAAALEwEAmpwYAAAGhElEQVR4nO2dXYgVZRjHf/8s98IkMkuXirSQwm5MNhOKPojKvLEgRC/KC2GDEoqIsC7K24K6CEKzsryoVCjLmz7Mgi4CP9Hygy0rpWTTPk2KPrSni/ddHbfdPbNnzpx90ucHw5kzZ84772/n48zs/M9zZGacCpwx0h1oFSHijRDxRog0QtIsST2S9kpaXNdyjmNmLR+AUcCXwKXAaGAHMLWOZfUNda2RGcBeM/vKzP4CVgFzaloWUN+mdSHwTeH5t3nagEiywvB9Mws8s5k3tQJJ3UD3AC/tb6a9ukQOABcXnl+Upx3HzJYDyyGtkaoLrGvT2gxMkTRZ0mhgHrCupmUBNa0RMzsqaRHwHukItsLMdtWxrD7k4TS+36a11cy6httGfLJ7I0S8ESLeCBFvhIg3QsQbIeKNEPFGiHgjRLwRIt4IEW+EiDdCxBsh4o0Q8UaIeKPSPURJ+4AjwDHgqJl1SRoHrAYmAfuAuWb2c7VuNqYVa+QmM5tWuO+3GNhgZlOADfl57dSxac0BVubxlcAdNSzjP1QVMeB9SVtzkgFggpn15vHvgAkDvVFSt6QtkrZU7EPuSbUU0IX58QJSAuh64Jd+8/xcoh0rDFvang4yswP58RCwlpQKOiipEyA/HqqyjLI0LSJpjKSxfePArcBOUlRjQZ5tAfB21U6WocrhdwKwVlJfO6+Z2buSNgNrJC0kJX3mVu9mYyLC4Y0Q8UaIeCNEvBEi3ggRb4SIN0LEGyHijRDxRoh4I0S8ESLeCBFvhIg3QsQbIeKNhiKSVkg6JGlnYdo4SeslfZEfz83TJenZXLDiU0nT6+x8kTJr5BVgVr9pg6Ubbgem5KEbWNqabpag5P30ScDOwvMeoDOPdwI9efx5YP5A83m9zz5YumFYRStaSeUKA2ZmzdRsGKJ4RVM0u0YGSzc0LFrRh5ktN7OuZm5FD0SzIoOlG9YB9+Sj10zgcGETrJcSO+LrQC/wN2mbXwicRzpafQF8AIzL8wp4jlRu5zOgq+TBpPLOHskHb4SIN0LEGyHijRDxRoh4I0S8ESLeCBFvhIg3QsQbIeKNEPFGiHgjRLwRIt4IEW+cPiKDRDiWSDogaXseZhdeezRHOHok3VZXx/9DiRuV1wPTOTn5sAR4eIB5p5IqDXQAk0k3RUe5SD6Y2cfATyX/LnOAVWb2p5l9DewlVR2onSr7yKKcAFrRlw5iGBGOVhevaFZkKXAZMI10D/7p4TbgIvlgZgfN7JiZ/QO8wInNp3SEo9U0JdKXQ8ncSSpaASnCMU9Sh6TJpNzWpmpdLEfDdJCk14EbgfGSvgWeAG6UNI10lNkH3AtgZrskrQF2A0eB+83sWC0979/PiHA4I0S8ESLeCBFvhIg3QsQbIeKNEPFGiHgjRLwRIt4IEW+EiDdCxBsh4o0Q8UaIeKNM8uFiSR9J2i1pl6QH8nRfBSxKpBI6gel5fCzwOSnh8BSwOE9fDDyZx2cD75C+Ej4T2NiO5MPw35C+hH8LLSxg0QqRYe0jkiYBVwEbcVbAonTxCklnA28AD5rZr/k3FYDmCliMSPEKSWeRJF41szfz5EoFLNoe4VD6078E7DGzZwov+SpgUWLnvo60E34KbM/DbFpYwIIoXnGC0+eT/f9CiHgjRLwRIt4IEW+EiDdCxBsh4o0Q8UaIeCNEvBEi3ggRb4SIN0LEGyHijRDxRpUIh68CFhUiHEtoUQEL2lS8otfMtuXxI8Aehk4yjEgBiyoRDqhQwGLEilf0j3BQsYBF25MPMHCEw5wVsGg6wuGtgEWZdNC1wN3AZ5K252mPAfM9FbDwEuH4HvgN+AG43MzGDrcNF5/sZnY+8EPe8XuaacOFSCsIkRpY3u9xWLjY2VuBpzVSibaKSJol6UtJv0s6WLwsyK93SFqdLw+O5kuH7ZIeb9h4M6fMzQzAKNIp/TWk05kdQBf5siDPcx+wjFQ9ahuwumWn8S1kBrDXzDaa2SZgFSnRXbwsmAOszOO9wM0qBoyHoJ0iA53eX8HJlwXFeWaSLuTWS7qyUeMjubN3kH7Rsu+yoMg24BKS7IvAW40aa6fI8dP7fFnwCLDZTiS7j8+Txf4AzgFWA2dJGj9U4+0U2QxMyaf2LwPnkmSKrAMWSJoI3AV8CFyd+/njUI239QMx/6dlGWnNHCR9y2Ei6asavaTLihtIR60xwH7gMPCQmX0yZNvxye6MEPFGiHgjRLxxyoj8C1ssfD4KBPuqAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.imshow(img)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
